master-firmware/src/robot_helpers/trajectory_helpers.c

```diff
@@ -125,6 +125,11 @@ static bool trajectory_is_cartesian(struct trajectory* traj)
     return traj->state == RUNNING_XY_START || traj->state == RUNNING_XY_ANGLE || traj->state == RUNNING_XY_ANGLE_OK || traj->state == RUNNING_XY_F_START || traj->state == RUNNING_XY_F_ANGLE || traj->state == RUNNING_XY_F_ANGLE_OK || traj->state == RUNNING_XY_B_START || traj->state == RUNNING_XY_B_ANGLE || traj->state == RUNNING_XY_B_ANGLE_OK;
 }
 
+bool trajectory_robot_inside_obstacle(struct _robot* robot, poly_t* opponent)
+{
+    return math_point_is_in_square(opponent, position_get_x_s16(&robot->pos), position_get_y_s16(&robot->pos));
```

Comment:
Name is misleading, is_in_square should be is_in_poly.

Reply:
true but here we assume for the sake of simplicity the opponent is a square